Snowplow/SPTracker.m

```diff
@@ -247,6 +248,48 @@ - (void) setSessionContext:(BOOL)sessionContext {
     }
 }
 
+- (void) setGDPRContextWithBasis:(SNOWProcessingBasis)basisForProcessing
+         withDocumentDescription:(NSString *)documentDescription
+             withDocumentVersion:(NSString *)documentVersion
+                  withDocumentId:(NSString *)documentId {
+    SPPayload * pl = [[SPPayload alloc] init];
+    [pl addValueToPayload:documentId forKey:@"documentId"];
+    [pl addValueToPayload:documentVersion forKey:@"documentVersion"];
+    [pl addValueToPayload:documentDescription forKey:@"documentDescription"];
+    NSString *pB = [[self class] getProcessingBasisString:basisForProcessing];
+    [pl addValueToPayload:pB forKey:@"basisForProcessing"];
+    SPSelfDescribingJson * sdj = [[SPSelfDescribingJson alloc] initWithSchema:kSPGDPRContextSchema andPayload:pl];
+    SNOWContext * sc = [[SNOWContext alloc] initWithContext:sdj];
+    [_globalContexts addContext:sc];
+}
+
++ (NSString*) getProcessingBasisString:(SNOWProcessingBasis) basis {
+    NSString *result = nil;
+    switch (basis) {
+        case SNOWProcessingBasisConsent:
+            result = @"consent";
+            break;
+        case SNOWProcessingBasisContract:
+            result = @"contract";
+            break;
+        case SNOWProcessingBasisLegalObligation:
+            result = @"legal_obligation";
+            break;
+        case SNOWProcessingBasisVitalInterest:
+            result = @"vital_interests";
+            break;
+        case SNOWProcessingBasisPublicTask:
+            result = @"public_task";
+            break;
+        case SNOWProcessingBasisLegitimateInterests:
+            result = @"legitimate_interests";
+            break;
+        default:
+            break;
+    }
+    return result;
```

Comment:
Is "nil" a valid result for the return result here?  Can `basisForProcessing` be `nil`?

Reply:
It's required.

Reply:
Does this now mean that if the result is null that it would then throw an exception due to the method return type?  Why can't we just have a default case for the switch that supplies a valid value?

Reply:
First part of the question: Yes (you get an exception)
What would be a sensible default in this case? Given that this would likely be a programmer omission probably throwing is the right thing to do (of course we would prefer a compile time error, so any ideas in that direction are welcome)